build/driver_test.go

```diff
@@ -226,7 +226,7 @@ func TestSelectNodeCurrentPlatform(t *testing.T) {
 	require.NoError(t, err)
 	require.True(t, perfect)
 	require.Len(t, res, 1)
-	require.Equal(t, "bbb", res[0].Node().Builder)
+	require.Equal(t, "aaa", res[0].Node().Builder)
```

Comment:
The real issue is that the platform in `opt` gets reset when it should not. We need to figure out a test to make sure it doesn't happen. Looks like the old build code only expect "resolveNode" to have a platform if it is an override to the existing platform but maybe this should be made safer (more explicit) somehow.

Reply:
Rebased on top of @jedevc changes in https://github.com/docker/buildx/compare/master...jedevc:buildx:fix-setting-wrong-platform-on-solve-opt to fix the solve opt case and remove the duplicated map.

Reply:
I guess you talk about the `TestSelectNodeNoPlatform` and default spec case right? It seems related to https://github.com/docker/buildx/issues/2119 and proposed new rules in https://github.com/docker/buildx/issues/2119#issuecomment-1815517521 should cover this in follow-up iiuc.